File: client/src/main/java/au/com/codeka/warworlds/client/opengl/DimensionResolver.java
package au.com.codeka.warworlds.client.opengl;

import android.content.Context;
import android.util.DisplayMetrics;
import android.util.TypedValue;

/**
 * Helper class for converting between various dimension units.
 */
public class DimensionResolver {
  private final DisplayMetrics displayMetrics;

  public DimensionResolver(Context context) {
    this.displayMetrics = context.getResources().getDisplayMetrics();
  }

  public float dp2px(float dp) {
    return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, displayMetrics);
  }
}

File: client/src/main/kotlin/au/com/codeka/warworlds/client/opengl/DimensionResolver.kt
package au.com.codeka.warworlds.client.opengl

import android.content.Context
import android.util.DisplayMetrics
import android.util.TypedValue

/**
 * Helper class for converting between various dimension units.
 */
class DimensionResolver(context: Context) {
  private val displayMetrics: DisplayMetrics = context.resources.displayMetrics

  fun dp2px(dp: Float): Float {
    return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, displayMetrics)
  }

  fun px2dp(px: Float): Float {
    return px / displayMetrics.density
  }
}